### PR TITLE
Add on error deployment ARM

### PR DESCRIPTION
**Pull request by chidmdxx**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by chidmdxx** [comment]
Shouldn't a new python sdk be needed too?

**Review by sergey-shandar — CHANGES_REQUESTED** [review]
[readme.md](https://github.com/chidmdxx/azure-rest-api-specs/tree/323d84050e2e245e2b8aa0ba5d2fd5d69b6aee8d/specification/resources/resource-manager) has to be changed as well

**Comment by sergey-shandar** [comment]
@ravbhatnagar new API version.

**Comment by ravbhatnagar** [comment]
Signing off from ARM side!

File: specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json [hunk 1]
```diff
@@ -0,0 +1,2955 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.Resources/deployments/{deploymentName}": {
```
Comment (sergey-shandar) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json]: Please, provide examples for operations.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json [hunk 2]
```diff
@@ -0,0 +1,2955 @@
...
+        }
+      },
+      "required": [
+        "uri"
+      ],
+      "description": "Entity representing the reference to the deployment paramaters."
+    },
+    "DeploymentProperties": {
+      "properties": {
+        "template": {
```
Comment (sergey-shandar) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json]: `name`, `id` and `type` has to be `readonly` properties.

Comment (sergey-shandar) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json]: `name`, `id` and `type` has to be `readonly` properties.

Comment (sergey-shandar) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json]: `name`, `id` and `type` has to be `readonly` properties.

**Review by sergey-shandar — CHANGES_REQUESTED** [review]

File: specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (chidmdxx) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json]: no other swagger for ARM has /operations api
and we don't have that implemented in frontdoor

Comment (chidmdxx) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json]: ARM doesn't have operations api

File: specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (chidmdxx) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json]: it shouldn't it's a deployment so it's different

[the remark this answers is not in the document]
Comment (chidmdxx) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json]: it shouldn't it's a deployment so it's different

Comment (chidmdxx) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json]: for on error deployment type should be assignable by the user

[the remark this answers is not in the document]
Comment (chidmdxx) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json]: which resource do you mean?

[the remark this answers is not in the document]
Comment (chidmdxx) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json]: which operation do you mean? there shouldn't be any new operations, just the change in the payload

**Comment by chidmdxx** [comment]
Error in CI syntax mode seems unrelated, let me know if I'm wrong about that assumption

'Error resolving $ref pointer "/home/travis/build/Azure/azure-rest-api-specs/specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json#/parameters/LoggerIdParameter". \nToken "LoggerIdParameter" does not exist.'

**Comment by chidmdxx** [comment]
is there a way to regen the python sdk? the 2018 version is not being auto created

**Comment by lmazuel** [comment]
@chidmdxx As the Python automation comment says, your PR is here:
https://github.com/Azure/azure-sdk-for-python/pull/2084

Thank you,

**Comment by sergey-shandar** [comment]
@chidmdxx I'm waiting for CI to see if there is no more issues.

**Comment by sergey-shandar** [comment]
@chidmdxx ruby and java CI are failing. We are investigating the issue.

**Comment by sergey-shandar** [comment]
@daschult please, have a look java CI.

**Comment by daschult** [comment]
@sergey-shandar I've identified the problem with the Java generator configuration for this pull request and I'm working on fixing it now.

File: specification/resources/resource-manager/readme.md [hunk 3]
```diff
@@ -41,7 +41,7 @@ tag: package-policy-2017-06
 ```
 
 ``` yaml $(package-resources)
-tag: package-resources-2017-05
```
Comment (sarangan12) [inline specification/resources/resource-manager/readme.md]: You can add a new tag. But, You cannot delete an existing tag,

**Comment by daschult** [comment]
@sergey-shandar I have a fix for the Java code generator problem in [this PR](https://github.com/Azure/azure-rest-api-specs/pull/2613).

**Comment by sergey-shandar** [comment]
@sarangan12 please, fix the configuration file in the ruby repository.

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**